src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx

```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                const connecting = new ProfileIdentifier(ui.networkIdentifier, username)
+                await Services.Identity.attachProfile(connecting, persona, { connectionConfirmState: 'confirmed' })
+                ui.taskGotoProfilePage(connecting)
+                setStep(SetupGuideStep.PasteIntoBio)
+                break
+            case SetupGuideStep.PasteIntoBio:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
+                    status: SetupGuideStep.SayHelloWorld,
+                    username,
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                ui.taskGotoNewsFeedPage()
+                setStep(SetupGuideStep.SayHelloWorld)
+                break
+            case SetupGuideStep.SayHelloWorld:
+                onClose()
+                break
+        }
     }
-    const onCancel = async () => {
-        const username_ = getUsername()
-        currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
-            status: 'during',
-            // ensure staying find-username page
-            username: '',
-            persona: persona.toText(),
-        } as ImmersiveSetupCrossContextStatus)
-        const connected = new ProfileIdentifier(ui.networkIdentifier, username_)
-        await Services.Identity.detachProfile(connected)
-        setStep(SetupGuideStep.FindUsername)
+    const onBack = async () => {
+        switch (step) {
+            case SetupGuideStep.PasteIntoBio:
+                const username_ = getUsername()
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
+                    status: SetupGuideStep.FindUsername,
+                    username: '', // ensure staying find-username page
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                const connected = new ProfileIdentifier(ui.networkIdentifier, username_)
+                await Services.Identity.detachProfile(connected)
+                setStep(SetupGuideStep.FindUsername)
+                break
+            case SetupGuideStep.SayHelloWorld:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
+                    ...lastState,
+                    status: SetupGuideStep.PasteIntoBio,
+                } as ImmersiveSetupCrossContextStatus)
+                const connecting = new ProfileIdentifier(ui.networkIdentifier, username)
+                ui.taskGotoProfilePage(connecting)
+                setStep(SetupGuideStep.PasteIntoBio)
+                break
+        }
+    }
+    const onPaste = async () => {
+        try {
+            await navigator.clipboard.writeText(provePost)
+            ui.taskPasteIntoBio(provePost)
+            setPastedStatus(true)
```

Comment:
If you need the paste result, maybe you want to use [useCopyToClipboard](https://github.com/streamich/react-use/blob/master/docs/useCopyToClipboard.md) in react-use libaray, it's cleaner.

Reply:
Not necessary here.